### PR TITLE
Fixes for windows and CI/DEV enhanncements 

### LINKED ISSUE #339: On Windows, "#writeStyle" tests fail
On Windows :
- When reading QML file, linebreaks are \r\n
- When writing style, linebreaks are \n only

To fix it, I suggest in function writeStyle of QGISStyleParser class

```
let renderOpts = { pretty: true, indent: '  ', newline: '\n' };
if ('win32' === process.platform) {
  renderOpts.newline = '\r\n'; 
}

return new Promise<WriteStyleResult<string>>(resolve => {
  try {
	const builder = new Builder({ renderOpts: renderOpts });
...
```

Is there a better way to do it ?

### LINKED ISSUE #340: build failed (on windows)
# Bug

npm run build fails on windows (not sure if it's related to os)

Error message is 
`
ERROR in ./src/QGISStyleParser.ts 30:5
Module parse failed: Unexpected token (30:5)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| import _get from 'lodash/get';
|
> type SymbolizerMap = {
|   [key: string]: Symbolizer[]
| };
 @ multi ./src/QGISStyleParser.ts main[0]
`

### PR DESCRIPTION
**Pull request by KaiVolland**
This adds several adaptions and improvements.

- Fixes #339 
- Fixes #340 
- Adds workflows to ensure everything works on windows and ubuntu
- Replaces depreacted tslint with eslint
- Transform all wellKnownName in the test data to lower case
- Adapts the npm scripts
- Updates all packages including webpack, babel and jest

### CODE REVIEW

**Comment by KaiVolland** [comment]
I hope this one will help developing on windows @pprev94 @slafayIGN. See also https://github.com/geostyler/geostyler-sld-parser/pull/660 and https://github.com/geostyler/geostyler-sld-parser/pull/659.

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    - name: Cache Node.js modules 💾
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.OS }}-node-
+          ${{ runner.OS }}-
+    - name: Install dependencies ⏬
```
Comment (jansule) [inline .github/workflows/on-pull-request.yml:31]: very minor: please add newline before step

File: .github/workflows/on-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      run: npm run test
+
+    - name: Build artifacts 🏗️
+      run: npm run build
+
+    - name: Publish to coveralls ⭐
+      # coverage/lcov.info was generated in the previous npm run build step
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jansule) [inline .github/workflows/on-pull-request.yml:47]: Please add newline

**Review by jansule — APPROVED** [review]
Very nice! Just some minor remarks you might want to address before merging

**Pull request merged**